### PR TITLE
Move function to disable input for second language proficiency to a watch method

### PR DESCRIPTION
**Pull request by dzldebby**
# Fix bug where input for teacher's second language proficiency is not disabled if there are multiple attempts to edit Second Language 

User input for teacher's second language proficiency is supposed to be disabled if value for second language is null. The existing method for this is included within the `checkSecondLanguage` method. This is not optimal because it does not account for multiple changes to the Second Language field.

## Moved the method to a `Watch` instead

This change implements essential features.

## Test Steps

1. Create a new Teacher
2. Second language proficiency should be disabled at the start
3. Type something in the Second Language field 
4. Second Language proficiency should be enabled 
5. Remove original input from Second Language and replace it with some other input 
6. Second Language proficiency should still be enabled

### CODE REVIEW

File: frontend/src/pages/NewTeacher.vue [hunk 1]
```diff
@@ -215,6 +215,13 @@ export default {
     file(val) {
       this.uploadFile();
     },
+
+    SecondLanguage: function (newVal, oldVal) {
+      console.log(this.SecondLanguage);
```
Comment (thchia) [inline frontend/src/pages/NewTeacher.vue]: You can remove this.

**Review by thchia — APPROVED** [review]

**Review by therealyingtong — APPROVED** [review]
Untested approval

**Review by nondescryptid — APPROVED** [review]
Tested - secondlanguage proficiency field remains enabled when i change the second language input.
To clarify, is it expected behaviour for SecondLanguage Proficiency to remain enabled even when I remove Second Language input (i.e. i put "Chinese" in, then delete it)? This is what's happening now 
@dzldebby

**Comment by therealyingtong** [comment]
@szktm The UI still registers "Chinese" as the Second Language even after you delete it. (To verify, look at the newly created teacher profile.)

I've opened #140 to address this. I think we can open another PR for that, and merge this PR as-is.

**Comment by dzldebby** [comment]
> Tested - secondlanguage proficiency field remains enabled when i change the second language input.
> To clarify, is it expected behaviour for SecondLanguage Proficiency to remain enabled even when I remove Second Language input (i.e. i put "Chinese" in, then delete it)? This is what's happening now
> @dzldebby

@szktm Thanks for spotting this! It's because I forgot to add the `else` case in the function. I've added that now, and also an additional line to reset whatever second language proficiency that had been selected back to `null`. So if you try to add a second language, and then delete it, what you should see is: 
1. Second Language Proficiency input is disabled again 
2. Second Language Proficiency is set to ""

**Comment by dzldebby** [comment]
> @szktm The UI still registers "Chinese" as the Second Language even after you delete it. (To verify, look at the newly created teacher profile.)
> 
> I've opened #140 to address this. I think we can open another PR for that, and merge this PR as-is.

@therealyingtong I think this is because of `this.selected.SecondLanguage` field was used instead of `this.SecondLanguage`. Once I replaced that, creating a new teacher worked as expected for me.

**Pull request merged**